pagopa/io-app
# chore: [LLK-37] Support key uniqueness on retries

## Short description
This PR support LolliPOP on retries. The implemented flow now fails during a login retry (which is blocked by single IdPs) since it uses the same public key.

## List of changes proposed in this pull request
- ts/features/lollipop/hooks/useLollipopLoginSource.tsx: changed input parameter to `loginUri?: string`, this will be useful on CIE login integration. Changed `taskGetPublicKey ` to `taskRegenerateKey` to generate a new key at every login request.
- ts/sagas/startup.ts: Restored lollipop key generation at startup to be able to have this information for already logged in users. 
- ts/sagas/startup/authenticationSaga.ts: remove key generation now made on startup and on every login request.
- ts/screens/authentication/IdpLoginScreen.tsx: exported to their own files some utility functions and types to be use on other components (e.g. on CIE login screen).
- ts/utils/crypto.ts: add a task to regenerate a key using the same `keyTag`.

## How to test
- Pointing to production, test the SPID login flow. 
- Try going back to IdP selection and reselect the same IdP and check that all works fine. 
- Wait 5 minutes on SPID login credential input form page to see the app generic error page. Tap on "Retry", all must work fine.
- Uninstall and reinstall the app and try all the above points.
- Check with some log that at every login the key is recreated but during the login flow it remains the same. 
- After a successful login check that the public key used for the login flow is the same to the one actually stored on the device. 
- Check that the key is the same as the one used to log in, even after an application restart (swipe close, and restart).
- With the dev server try disabling lollipop and check that the login flow falls back to the standard one.
